src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -83,24 +88,58 @@ public SparkHeaderlessBAMOutputFormat() {
         }
     }
 
+    // Output format class for writing CRAM files through saveAsNewAPIHadoopFile. Must be public.
+    public static class SparkCRAMOutputFormat extends KeyIgnoringCRAMOutputFormat<NullWritable> {
```

Comment:
Its only because the two classes have different base classes that live in Hadoop-BAM. I don't see any useful way to consolidate them here, though going forward we may want to change AnySAMOutputFormat in Hadoop-BAM to delegate based on file extension similar to the way AnySAMInputFormat does.